packages/ember-application/lib/system/application.js

```diff
@@ -456,50 +577,74 @@ var Application = Namespace.extend(RegistryProxy, {
   },
 
   /**
-    Calling initialize manually is not supported.
-
-    Please see Ember.Application#advanceReadiness and
-    Ember.Application#deferReadiness.
+    Initialize the application and return a promise that resolves with the `Ember.Application`
+    object when the boot process is complete.
 
-    @private
-    @deprecated
-    @method initialize
-   **/
-  initialize() {
-    deprecate('Calling initialize manually is not supported. Please see Ember.Application#advanceReadiness and Ember.Application#deferReadiness');
-  },
+    Run any application initializers and run the application load hook. These hooks may
+    choose to defer readiness. For example, an authentication hook might want to defer
+    readiness until the auth token has been retrieved.
 
-  /**
-    Initialize the application. This happens automatically.
+    By default, this method is called automatically on "DOM ready"; however, if autoboot
+    is disabled, you would need to call this method yourself. Calling this method before
+    "DOM ready" is not recommended.
```

Comment:
They had a streaming (as in HTTP chucked response) setup that keeps the body tag open for sending data (<script> ...json... </script>), which causes the DOM ready to fire very late, which delayed ember boot for no reason.

The actual caveats are rather nuanced:

1. when you kick off the boot, you must already have everything you need for your app, e.g. if your initializer and a route uses some global Google Analytics or Intercom thing, you must not boot it before they are loaded

2. When Ember eventually inserts the top level view (`didCreateRootView`), the rootElement must already be on the page

It is actually not that easy to determine these two things, and for a normal set up the gain is very little, so I think it's too nuanced to describe and recommend publicly in the docs.

However, for @chadhietala's use case, it is not that hard to determine "would have been dom ready if not for the streaming setup" point – you can just insert a synchronous script tag right before you start streaming data, so I was going to make sure that there is a thing you can call to tell Ember this is the "effective" DOM ready. Perhaps I'll just rename _autoBoot back to DOM ready or something

(I don't think we need to document this)